Source: http://sourceware.org/bugzilla/show_bug.cgi?id=12097
Upstream-Status: Submitted

The patch should be merged into glibc-ports

-Khem
Index: libc/ports/sysdeps/unix/sysv/linux/arm/eabi/libc-do-syscall.S
===================================================================
--- libc.orig/ports/sysdeps/unix/sysv/linux/arm/eabi/libc-do-syscall.S
+++ libc/ports/sysdeps/unix/sysv/linux/arm/eabi/libc-do-syscall.S
@@ -16,6 +16,8 @@
    Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
    02111-1307 USA.  */
 
+#if defined(__thumb__)
+
 #include <sysdep.h>
 
 /* Out-of-line syscall stub.  We expect the system call number in ip
@@ -41,3 +43,5 @@ ENTRY (__libc_do_syscall)
 	pop	{r7, pc}
 	.fnend
 END (__libc_do_syscall)
+
+#endif /* __thumb__ */
